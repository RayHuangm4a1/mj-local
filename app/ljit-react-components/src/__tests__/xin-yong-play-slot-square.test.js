import React from 'react';
import { shallow, mount, } from 'enzyme';
import XinYongPlaySlotSquare from '../components/xin-yong-play-slot/square';

const { TypeEnums, } = XinYongPlaySlotSquare;

describe('XinYongPlaySlotSquare', () => {
	it('handle default props', () => {
		const {
			className,
			playSlotType,
			play,
			isSelected,
			isDisabled,
			onClick,
		} = XinYongPlaySlotSquare.defaultProps;

		expect(className).toEqual('');
		expect(playSlotType).toEqual('');
		expect(play).toMatchObject({});
		expect(isSelected).toEqual(false);
		expect(isDisabled).toEqual(false);
		expect(onClick).toBeDefined();
		expect(onClick).toBeInstanceOf(Function);
	});

	it('should render correctly', () => {
		const props = {
			className: 'mock-class',
			play: { id: 0, name: 0, odds: 1.995, },
			isSelected: false,
			playSlotType: '',
			isDisabled: false,
			onClick: () => {},
		};

		const wrapper = shallow(<XinYongPlaySlotSquare {...props}/>);

		expect(wrapper).toMatchSnapshot();
	});

	it ('should be selectable by class ljit-xin-yong-play-slot-square', () => {
		const props = {
			play: { id: 0, name: 0, odds: 1.995, },
		};
		const wrapper = shallow(<XinYongPlaySlotSquare {...props} />);

		expect(wrapper.hasClass('ljit-xin-yong-play-slot-square')).toEqual(true);
	});

	it ('should be selectable by custom class', () => {
		const props = {
			play: { id: 0, name: 0, odds: 1.995, },
			className: 'mock-class',
		};
		const wrapper = shallow(<XinYongPlaySlotSquare {...props} />);

		expect(wrapper.hasClass('mock-class')).toEqual(true);
	});

	it ('should be selectable by class ljit-xin-yong-play-slot-square--disabled, if isDisabled is pass true', () => {
		const props = {
			play: { id: 0, name: 0, odds: 1.995, },
			isDisabled: true,
		};
		const wrapper = shallow(<XinYongPlaySlotSquare {...props} />);

		expect(wrapper.hasClass('ljit-xin-yong-play-slot-square--disabled')).toEqual(true);
	});

	it ('should be selectable by class ljit-xin-yong-play-slot-square--selected, if isSelected is pass true', () => {
		const props = {
			play: { id: 0, name: 0, odds: 1.995, },
			isSelected: true,
		};
		const wrapper = shallow(<XinYongPlaySlotSquare {...props} />);

		expect(wrapper.hasClass('ljit-xin-yong-play-slot-square--selected')).toEqual(true);
	});

	it('should handle onClick event', () => {
		const onClick = jest.fn();

		const wrapper = shallow(
			<XinYongPlaySlotSquare
				play={{ id: 0, name: 0, odds: 1.995, }}
				onClick={onClick}
			/>
		);

		wrapper.simulate('click');

		expect(onClick).toBeDefined();
		expect(onClick).toHaveBeenCalledTimes(1);
	});

	it ('should mount in a full DOM', () => {
		const className ='mock-class';
		const play = { id: 0, name: 0, odds: 1.995, };
		const isSelected = false;
		const isDisabled = false;
		const playSlotType = '';
		const onClick = () => {};
		const wrapper = mount(
			<XinYongPlaySlotSquare
				className={className}
				play={play}
				isSelected={isSelected}
				isDisabled={isDisabled}
				playSlotType={playSlotType}
				onClick={onClick}
			/>
		);

		expect(wrapper.props().className).toEqual(className);
		expect(wrapper.props().play).toMatchObject(play);
		expect(wrapper.props().isSelected).toEqual(isSelected);
		expect(wrapper.props().isDisabled).toEqual(isDisabled);
		expect(wrapper.props().playSlotType).toEqual(playSlotType);
		expect(wrapper.props().onClick).toEqual(onClick);
	});
});

describe('XinYongPlaySlotSquare TypeEnums', () => {
	it('should contains CIRCLE property', () => {
		const typeName = 'circle';
		const formatType = 'CIRCLE';

		expect(TypeEnums[formatType]).toEqual(typeName);
	});

	it('should contains ROUND property', () => {
		const typeName = 'round';
		const formatType = 'ROUND';

		expect(TypeEnums[formatType]).toEqual(typeName);
	});

	it('should contains RECTANGLE property', () => {
		const typeName = 'rectangle';
		const formatType = 'RECTANGLE';

		expect(TypeEnums[formatType]).toEqual(typeName);
	});

	it('should contains ANIMAL property', () => {
		const typeName = 'animal';
		const formatType = 'ANIMAL';

		expect(TypeEnums[formatType]).toEqual(typeName);
	});
});
