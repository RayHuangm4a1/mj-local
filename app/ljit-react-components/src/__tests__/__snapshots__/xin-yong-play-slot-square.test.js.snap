// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XinYongPlaySlotSquare should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <XinYongPlaySlotSquare
    className="mock-class"
    isDisabled={false}
    isSelected={false}
    onClick={[Function]}
    play={
      Object {
        "id": 0,
        "name": 0,
        "odds": 1.995,
      }
    }
    playSlotType=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ljit-xin-yong-play-slot-square__ball"
        >
          0
        </div>,
        <div
          className="ljit-xin-yong-play-slot-square__odds"
        >
          <DecimalNumber
            className=""
            data={1.995}
            decimalPlaces={4}
            hasSeparator={true}
            isCurrency={false}
            isPercent={false}
            isTolerance={false}
          />
        </div>,
      ],
      "className": "ljit-xin-yong-play-slot-square mock-class",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0,
          "className": "ljit-xin-yong-play-slot-square__ball",
        },
        "ref": null,
        "rendered": 0,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <DecimalNumber
            className=""
            data={1.995}
            decimalPlaces={4}
            hasSeparator={true}
            isCurrency={false}
            isPercent={false}
            isTolerance={false}
          />,
          "className": "ljit-xin-yong-play-slot-square__odds",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "",
            "data": 1.995,
            "decimalPlaces": 4,
            "hasSeparator": true,
            "isCurrency": false,
            "isPercent": false,
            "isTolerance": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ljit-xin-yong-play-slot-square__ball"
          >
            0
          </div>,
          <div
            className="ljit-xin-yong-play-slot-square__odds"
          >
            <DecimalNumber
              className=""
              data={1.995}
              decimalPlaces={4}
              hasSeparator={true}
              isCurrency={false}
              isPercent={false}
              isTolerance={false}
            />
          </div>,
        ],
        "className": "ljit-xin-yong-play-slot-square mock-class",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0,
            "className": "ljit-xin-yong-play-slot-square__ball",
          },
          "ref": null,
          "rendered": 0,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <DecimalNumber
              className=""
              data={1.995}
              decimalPlaces={4}
              hasSeparator={true}
              isCurrency={false}
              isPercent={false}
              isTolerance={false}
            />,
            "className": "ljit-xin-yong-play-slot-square__odds",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "data": 1.995,
              "decimalPlaces": 4,
              "hasSeparator": true,
              "isCurrency": false,
              "isPercent": false,
              "isTolerance": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
